Source/SwiftLintFramework/Reporters/JUnitReporter.swift

```diff
@@ -19,11 +19,12 @@ public struct JUnitReporter: Reporter {
     public static func generateReport(_ violations: [StyleViolation]) -> String {
         return "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<testsuites><testsuite>" +
             violations.map({ violation in
-                let fileName = violation.location.file ?? "<nopath>"
+                let fileName = violation.location.file?.escapedForXml() ?? "<nopath>"
```

Comment:
`"<nopath>"` needs to be escaped: `(violation.location.file ?? "<nopath>").escapedForXml()`